src/modules/admin/components/users/UserAuditHistory.tsx

```diff
@@ -0,0 +1,113 @@
+import { Stack, Typography } from '@mui/material';
+import { filter } from 'lodash-es';
+import {
+  ContextualCollapsibleList,
+  ContextualListExpansionButton,
+} from '@/components/elements/CollapsibleListContext';
+import { ColumnDef } from '@/components/elements/table/types';
+import useSafeParams from '@/hooks/useSafeParams';
+import AuditObjectChangesSummary, {
+  ObjectChangesType,
+} from '@/modules/audit/components/AuditObjectChangesSummary';
+import GenericTableWithData from '@/modules/dataFetching/components/GenericTableWithData';
+import { hasMeaningfulValue } from '@/modules/form/util/formUtil';
+import RelativeDateTableCellContents from '@/modules/hmis/components/RelativeDateTableCellContents';
+import { auditActionForDisplay } from '@/modules/hmis/hmisUtil';
+import {
+  UserAuditEventFilterOptions,
+  GetUserAuditEventsDocument,
+  GetUserAuditEventsQuery,
+  GetUserAuditEventsQueryVariables,
+} from '@/types/gqlTypes';
+
+type AuditHistoryType = NonNullable<
```

Comment:
There's a lot of duplicated code here across `ClientAuditHistory`, `EnrollmentAuditHistory`, and  now `UserAuditHistory`. It can't be fully factored out because of typescript -- they each use a different `Get<Something>AuditEventsQuery` in the `AuditHistoryType` -- but maybe it could be factored out in smaller ways, such as centrally defining the contents of the items in `columns`?

Reply:
Ok, I took a stab at reducing repetition but with a couple open Q's left in todos in the `auditHistoryColumnDefs.tsx` file.
1. maybe there would be a way to achieve this 'upstream', as it were, by using a smarter graphql fragment?
2. relatedly: the changes I've made still contain some repetitive code, it's just repetitive in a different way -- maybe this is better (because we can now easily change the code for any audit history table, including formatting/styling, without changing it in 3 places), but maybe it's actually worse and could lead to more confusion later on (because updates to the graphql schema for audit history types won't be reflected in the manually defined interface)? What do you think?

Reply:
With the more recent changes I'm especially unsure of this refactor -- please let me know if you think I should revert it or keep thinking about some other way the code could be more dry and less brittle!